### ISSUE TITLE
config issue

### ISSUE THREAD

**Issue by cfoellmann** [DannyBen/redirectly#10]
Hi @DannyBen 

I  know you can not give support for this great container but maybe it is very easy for you to correct.
I am trying to create this redirect:

- source: domain.de
- include all subdomains *.domain.de
- preserve full "path" *.domain.de/folder/?something=value

Trying: 
`*.domain.de/:anything = https://domain.com/%{anything}`

only this works:

`https://www.domain.de/dsfdasfAF` -- result: https://domain.com/dsfdasfAF

All others are not working:
`https://domain.de/dsfdasfAF`
`https://www.domain.de/dsfdasfAF/?this=that`
`https://www.domain.de`
`https://domain.de`

my test domain is:

ecocyte.de --> ecocyte-us.com

**Comment by DannyBen** [comment]
> I know you can not give support for this great container

Says who? 😉 

---

Your expectation for this use case is justified. It should work.
The pattern should be something like this:

```
(*)localhost/*fullpath
```

and in redirectly's INI file:

```ini
(*)localhost/*fullpath = https://example.com/%{fullpath}
```

The `*fullpath` matches anything, including slashes, as stated by the [mustermann documentation](https://github.com/sinatra/mustermann/blob/main/mustermann/README.md#sinatra) (which redirectly is based on).

The problem I see in the code now is that I escape all values (`fullpath`), so you currently end up getting `%2F` instead of the slash (`https://example.com/1%2F2`).

https://github.com/DannyBen/redirectly/blob/1fc886c494997b8d307a91b221325ec068da6096/lib/redirectly/app.rb#L75

I need to see why I did that, and if it is necessary. Let me do some more digging.

**Comment by DannyBen** [comment]
I have pushed a new docker image tagged [`edge`](https://hub.docker.com/r/dannyben/redirectly/tags) with a fix that should allow you to do this now.

In your redirects.ini put something like this:

```ini
(*)localhost/*rest = https://example.com/%{rest}
```

Pattern explanation:

| Token | Meaning
|--|--
| `(*)`  |  wildcard, match anything. The parentheses are needed to avoid confusing it with `*localhost` which means something else (a splat named `localhost`).
| `*rest` |  match everything, including slashes, and make it available as `%{rest}` in the target URL


Let me know if this is what you need, and I will release a new version.

**Comment by cfoellmann** [comment]
Wow!

Thanks you so much and with this verbosity in your communication!

The edge version does exactly what I described and you have implemented.

**Comment by DannyBen** [comment]
Excellent. I will release a version later today.
Thanks for opening this issue and feel free to ask if you have any other problem or request.

**Comment by DannyBen** [comment]
This is now fixed and released in version 0.3.0.
